plugin/opentelemetry/setup.go

```diff
@@ -0,0 +1,165 @@
+		case 1:
+			ot.EndpointType, ot.Endpoint, err = normalizeEndpoint(defEpType, args[0])
+		case 2:
+			epType := strings.ToLower(args[0])
+			ot.EndpointType, ot.Endpoint, err = normalizeEndpoint(epType, args[1])
+		default:
+			err = c.ArgErr()
+		}
+		if err != nil {
+			return ot, err
+		}
+
+		ot.batchSpanOptions = trace.BatchSpanProcessorOptions{
+			MaxQueueSize:       2048,
+			BatchTimeout:       5 * time.Second,
+			ExportTimeout:      30 * time.Second,
+			MaxExportBatchSize: 512,
+		}
+		ot.SamplingProbability = 1
+		ot.Insecure = false
+
+		for c.NextBlock() {
+			switch c.Val() {
+			case "service":
+				args := c.RemainingArgs()
+				if len(args) != 1 {
+					return nil, c.ArgErr()
+				}
+				ot.serviceName = args[0]
+			case "max_queue_size":
+				args := c.RemainingArgs()
+				if len(args) != 1 {
+					return nil, c.ArgErr()
+				}
+				if ot.batchSpanOptions.MaxQueueSize, err = strconv.Atoi(args[0]); err != nil {
+					return ot, err
+				}
+			case "batch_timeout":
+				args := c.RemainingArgs()
+				if len(args) != 1 {
+					return nil, c.ArgErr()
+				}
+				if ot.batchSpanOptions.BatchTimeout, err = time.ParseDuration(args[0]); err != nil {
+					return nil, err
+				}
+			case "export_timeout":
+				args := c.RemainingArgs()
+				if len(args) != 1 {
+					return nil, c.ArgErr()
+				}
+				if ot.batchSpanOptions.ExportTimeout, err = time.ParseDuration(args[0]); err != nil {
+					return nil, err
+				}
+			case "max_export_batch_size":
+				args := c.RemainingArgs()
+				if len(args) != 1 {
+					return nil, c.ArgErr()
+				}
+				if ot.batchSpanOptions.MaxExportBatchSize, err = strconv.Atoi(args[0]); err != nil {
+					return nil, err
+				}
+			case "sampling_probability":
+				args := c.RemainingArgs()
+				if len(args) != 1 {
+					return nil, c.ArgErr()
+				}
+				if ot.SamplingProbability, err = strconv.ParseFloat(args[0], 64); err != nil {
+					return nil, err
+				}
+				if ot.SamplingProbability < 0 || ot.SamplingProbability > 1 {
+					return nil, fmt.Errorf("sampling probability needs to be in ragne 0<=n<=1")
+				}
+			case "insecure":
+				args := c.RemainingArgs()
+				if len(args) != 0 {
+					return nil, c.ArgErr()
+				}
+				ot.Insecure = true
+			default:
+				log.Warningf("unknown argument while parsing configuration '%s'", c.Val())
+			}
+		}
+	}
+
+	return ot, err
+}
+
+func normalizeEndpoint(epType, ep string) (string, string, error) {
+	if _, ok := supportedProviders[epType]; !ok {
+		return "", "", fmt.Errorf("tracing endpoint type '%s' is not supported", epType)
+	}
+
+	if ep == "" {
+		ep = supportedProviders[epType]
+	}
+
+	if epType == "zipkin" {
+		if !strings.Contains(ep, "http") {
+			ep = "http://" + ep + "/api/v2/spans"
+		}
+	} else if epType == "otelhttp" {
+		if strings.Contains(ep, "http") || strings.Contains(ep, "/") {
+			return "", "", fmt.Errorf("invalid format of opentelemetry endpoint: '%s', valid format: 'HOST:PORT'", ep)
```

Comment:
I think we should use the official `OpenTelemetry` name with uppercase O and T.

Reply:
yea, seems better, fixed!